velox/vector/BaseVector.cpp

```diff
@@ -974,7 +986,7 @@ BufferPtr BaseVector::sliceBuffer(
   if (offset % 8 == 0) {
     return sliceBufferZeroCopy(1, true, buf, offset / 8, (length + 7) / 8);
   }
-  VELOX_DCHECK_NOT_NULL(pool);
+  VELOX_CHECK_NOT_NULL(pool, "Pool must not be null.");
```

Comment:
It used to be a DCHECK? Is there any impact on perf?

Reply:
I think this would not make much perf difference, considering the memory allocation and data copying after the check.